File: leetcode2/1easy/최원준/Q2970.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q2970 {
    class Solution {
        int[] nums;
        int n;
        public boolean check(int start, int end) {
            int prev = Integer.MIN_VALUE;
            for (int i=0; i<n; i++) {
                if (start <= i && i <= end) continue;
                if (prev >= nums[i]) return false;
                prev = nums[i];
            }
            return true;
        }

        public int incremovableSubarrayCount(int[] nums) {
            this.nums = nums;
            n = nums.length;

            int ans = 0;
            for (int i=0; i<n; i++) for (int j=i; j<n; j++) {
                if (check(i,j)) ans++;
            }
            return ans;
        }
    }
}

File: leetcode2/1easy/강병호/Q2970.py
# from itertools import combinations

# class Leetcode.이재훈.이재훈.Solution:
#     def incremovableSubarrayCount(self, nums: list[int]) -> int:
#         # for i in range(1, len(nums)+1, 1): # subarray lens
#         right = len(nums) - 1
#         left = 0

#         while left < 
#             newList = list()
#             # 0~start + end ~ last
#             newList = nums[0:start] + nums[end:]
#             print(f"newList : {newList}")
#             for j in range(0, len(newList)-1):
#                 if newList[j] >= newList[j+1]:
#                     break
#                 if j == len(newList) - 1:
#                     print("good")
#                     result+=1
#         return result


# if __name__ == "__main__":
#     sol = Leetcode.이재훈.이재훈.Solution()
#     print(sol.incremovableSubarrayCount([1, 2, 3, 4]))
#     print(sol.incremovableSubarrayCount([8, 7, 6, 6]))